vaadin-text-field-flow-parent/vaadin-text-field-flow/src/main/java/com/vaadin/flow/component/textfield/AbstractNumberField.java

```diff
@@ -229,11 +230,38 @@ public void setHasControls(boolean hasControls) {
      * @see #setStep(double)
      *
      * @return {@code true} if buttons are visible, {@code false} otherwise
+     * @deprecated since 23.3. Use {@link #isStepButtonsVisible()} instead.
      */
+    @Deprecated
     public boolean hasControls() {
         return super.hasControlsBoolean();
     }
 
+    /**
+     * Sets the visibility of the buttons for increasing/decreasing the value
+     * accordingly to the default or specified step.
+     *
+     * @see #setStep(double)
+     *
+     * @param stepButtonsVisible
+     *            {@code true} if control buttons should be visible;
+     *            {@code false} if those should be hidden
+     */
+    public void setStepButtonsVisible(boolean stepButtonsVisible) {
+        getElement().setProperty("hasControls", stepButtonsVisible);
+    }
+
+    /**
+     * Gets whether the buttons for increasing/decreasing the value are visible.
+     *
+     * @see #setStep(double)
+     *
+     * @return {@code true} if buttons are visible, {@code false} otherwise
+     */
+    public boolean isStepButtonsVisible() {
```

Comment:
Using singular here ("is") as it is the common naming convention for boolean getters.